command/generate-root.go

```diff
@@ -0,0 +1,348 @@
+	// Check the status
+	status, err := client.Sys().RootGenerationStatus()
+	if err != nil {
+		c.Ui.Error(fmt.Sprintf("Error reading root generation status: %s", err))
+		return 1
+	}
+
+	c.dumpStatus(status)
+
+	return 0
+}
+
+// dumpStatus dumps the status to output
+func (c *GenerateRootCommand) dumpStatus(status *api.RootGenerationStatusResponse) {
+	// Dump the status
+	statString := fmt.Sprintf(
+		"Nonce: %s\n"+
+			"Started: %v\n"+
+			"Rekey Progress: %d\n"+
+			"Required Keys: %d\n"+
+			"Complete: %t",
+		status.Nonce,
+		status.Started,
+		status.Progress,
+		status.Required,
+		status.Complete,
+	)
+	if len(status.PGPFingerprint) > 0 {
+		statString = fmt.Sprintf("%s\nPGP Fingerprint: %s", statString, status.PGPFingerprint)
+	}
+	if len(status.EncodedRootToken) > 0 {
+		statString = fmt.Sprintf("%s\n\nEncoded root token: %s", statString, status.EncodedRootToken)
+	}
+	c.Ui.Output(statString)
+}
+
+func (c *GenerateRootCommand) Synopsis() string {
+	return "Promotes a token to a root token"
+}
+
+func (c *GenerateRootCommand) Help() string {
+	helpText := `
+Usage: vault generate-root [options] [key]
+
+  'generate-root' is used to create a new root token.
+
+  Root generation can only be done when the Vault is already unsealed. The
+  operation is done online, but requires that a threshold of the current unseal
+  keys be provided.
+
+  One (and only one) of the following must be provided at attempt
+  initialization time:
+  
+  1) A 16-byte, base64-encoded One Time Password (OTP) provided in the '-otp'
+  flag; the token is XOR'd with this value before it is returned once the final
+  unseal key has been provided. The '-decode' operation can be used with this
+  value and the OTP to output the final token value. The '-genotp' flag can be
+  used to generate a suitable value.
+
+  or
+
+  2) A file containing a PGP key (binary or base64-encoded) or a Keybase.io
+  username in the format of "keybase:<username>" in the '-pgp-key' flag. The
+  final token value will be encrypted with this public key and base64-encoded.
+  
+General Options:
+
+  ` + generalOptionsUsage() + `
+
+Rekey Options:
+
+  -init                   Initialize the root generation attempt. This can only
+                          be done if no generation is already initiated.
+
+  -cancel                 Reset the root generation process by throwing away
+                          prior unseal keys and the configuration.
+
+  -status                 Prints the status of the current attempt. This can be
+                          used to see the status without attempting to provide
+                          an unseal key.
+
+  -decode=abcd            Decodes and outputs the generated root token. The OTP
+                          used at '-init' time must be provided in the '-otp'
+                          parameter.
+
+  -genotp                 Returns a high-quality OTP suitable for passing into
```

Comment:
Probably even better would be to show the help text? Otherwise they may not understand what they're even being shown.